pkg/cloud/gcp/actuators/util/gcp_tags_labels_test.go

```diff
@@ -252,3 +368,279 @@ func TestGetLabelsList(t *testing.T) {
+				return infra
+			},
+			machineSpecTags: []machinev1.ResourceManagerTag{
+				{ParentID: "openshift", Key: "key1", Value: "value1"},
+				{ParentID: "openshift", Key: "key2", Value: "value2"},
+			},
+			expectedTags: map[string]string{
+				"tagKeys/0076293608": "tagValues/0055629312", "tagKeys/0023033386": "tagValues/0088842980",
+			},
+		},
+		{
+			name:               "user-defined tags feature is disabled",
+			tagsFeatureEnabled: false,
+			getInfra: func() *configv1.Infrastructure {
+				infra := new(configv1.Infrastructure)
+				infraObj.DeepCopyInto(infra)
+				infra.Status.PlatformStatus.GCP.ResourceTags = []configv1.GCPResourceTag{
+					{ParentID: "openshift", Key: "key1", Value: "value1"},
+					{ParentID: "openshift", Key: "key2", Value: "value2"},
+				}
+				return infra
+			},
+			machineSpecTags: []machinev1.ResourceManagerTag{
+				{ParentID: "openshift", Key: "key2", Value: "value2"},
+				{ParentID: "openshift", Key: "key3", Value: "value3"},
+			},
+			expectedTags: nil,
+		},
+		{
+			name:               "user-defined tags feature is enabled",
+			tagsFeatureEnabled: true,
+			getInfra: func() *configv1.Infrastructure {
+				infra := new(configv1.Infrastructure)
+				infraObj.DeepCopyInto(infra)
+				infra.Status.PlatformStatus.GCP.ResourceTags = []configv1.GCPResourceTag{
+					{ParentID: "openshift", Key: "key1", Value: "value1"},
+					{ParentID: "openshift", Key: "key2", Value: "value2"},
+				}
+				return infra
+			},
+			machineSpecTags: []machinev1.ResourceManagerTag{
+				{ParentID: "openshift", Key: "key3", Value: "value3"},
+				{ParentID: "openshift", Key: "key4", Value: "value4"},
+			},
+			expectedTags: map[string]string{
+				"tagKeys/0076293608": "tagValues/0055629312",
+				"tagKeys/0023033386": "tagValues/0088842980",
+				"tagKeys/0069899625": "tagValues/0047417661",
+				"tagKeys/0050205103": "tagValues/0036307885",
+			},
+		},
+		{
+			name:               "providerSpec tags is empty",
+			tagsFeatureEnabled: true,
+			getInfra: func() *configv1.Infrastructure {
+				infra := new(configv1.Infrastructure)
+				infraObj.DeepCopyInto(infra)
+				infra.Status.PlatformStatus.GCP.ResourceTags = []configv1.GCPResourceTag{
+					{ParentID: "openshift", Key: "key1", Value: "value1"},
+					{ParentID: "openshift", Key: "key2", Value: "value2"},
+				}
+				return infra
+			},
+			machineSpecTags: []machinev1.ResourceManagerTag{},
+			expectedTags: map[string]string{
+				"tagKeys/0076293608": "tagValues/0055629312",
+				"tagKeys/0023033386": "tagValues/0088842980",
+			},
+		},
+		{
+			name:               "providerSpec tags maxes allowed limit",
+			tagsFeatureEnabled: true,
+			getInfra: func() *configv1.Infrastructure {
+				infra := new(configv1.Infrastructure)
+				infraObj.DeepCopyInto(infra)
+				return infra
+			},
+			machineSpecTags: []machinev1.ResourceManagerTag{
+				{ParentID: "openshift", Key: "key1", Value: "value1"}, {ParentID: "openshift", Key: "key2", Value: "value2"},
```

Comment:
nit: storing these in a `const` or a lambda and referring to them may make readability better (and avoid some duplication)

Reply:
suggestion incorporated.